src/drop-down/drop-down.component.ts

```diff
@@ -0,0 +1,375 @@
+     * user will be able to use keyboard navigation.
+     */
+    @Input()
+    public allowItemsFocus = true;
+
+    /**
+     * Gets/sets the drop down's id
+     */
+    @Input()
+    get id(): string {
+        return this._id;
+    }
+    set id(value: string) {
+        this._id = value;
+        this.toggleDirective.id = value;
+    }
+
+    /**
+     * Get currently selected item
+     */
+    public get selectedItem(): IgxDropDownItemComponent {
+        const selection = this.selectionAPI.get_selection(this.id);
+        return selection && selection.length > 0 ? selection[0] as IgxDropDownItemComponent : null;
+    }
+
+    /**
+     * Get all non-header items
+     */
+    public get items(): IgxDropDownItemComponent[] {
+        const items: IgxDropDownItemComponent[] = [];
+        if (this.children !== undefined) {
+            for (const child of this.children.toArray()) {
+                if (!child.isHeader) {
+                    items.push(child);
+                }
+            }
+        }
+
+        return items;
+    }
+
+    /**
+     * Get all header items
+     */
+    public get headers(): IgxDropDownItemComponent[] {
+        const headers: IgxDropDownItemComponent[] = [];
+        if (this.children !== undefined) {
+            for (const child of this.children.toArray()) {
+                if (child.isHeader) {
+                    headers.push(child);
+                }
+            }
+        }
+
+        return headers;
+    }
+
+    /**
+     * Get dropdown html element
+     */
+    public get element() {
+        return this.elementRef.nativeElement;
+    }
+
+    constructor(
+        private elementRef: ElementRef,
+        private cdr: ChangeDetectorRef,
+        private selectionAPI: IgxSelectionAPIService) { }
+
+    /**
+     * Select an item by index
+     * @param index of the item to select
+     */
+    setSelectedItem(index: number) {
+        if (index < 0 || index >= this.items.length) {
+            return;
+        }
+
+        const newSelection = this.items.find((item) => item.index === index);
+        if (newSelection.isDisabled) {
+            return;
+        }
+
+        this.changeSelectedItem(newSelection);
+    }
+
+    /**
+     * Opens the dropdown
+     */
+    open(fireEvents?: boolean) {
+        this.toggleDirective.open(true);
+    }
+
+    /**
+     * Closes the dropdown
+     */
+    close(fireEvents?: boolean) {
+        this.toggleDirective.close(true);
+    }
+
+    /**
+     * Toggles the dropdown
+     */
+    toggle(fireEvents?: boolean) {
+        if (this.toggleDirective.collapsed) {
+            this.open();
+        } else {
+            this.close();
+        }
+    }
+
+    focusFirst() {
+        let focusedItemIndex = 0;
+        while (this.items[focusedItemIndex] && this.items[focusedItemIndex].isDisabled) {
+            focusedItemIndex++;
+        }
+        if (focusedItemIndex < this.items.length - 1) {
```

Comment:
We have a little code repetition here and in focusLast. Please refactor

Reply:
Done